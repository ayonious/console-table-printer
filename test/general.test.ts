import { Table } from '../index';

describe('General tests', () => {
  it('should yield the same state when adding rows during instantation than with method', () => {
    const rows = [{ foo: '1', bar: '2' }];

    const table1 = new Table();
    table1.addRows(rows);

    const table2 = new Table({ rows });

    expect(table1.table).toStrictEqual(table2.table);
  });
});
