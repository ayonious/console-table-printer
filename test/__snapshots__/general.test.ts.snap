// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Table Functionality should handle empty table creation 1`] = `
"┐
┤
┘"
`;

exports[`General Table Functionality should handle mixed data types in cells 1`] = `
"┌────────┬────────┬─────────┬───────────┬────────────────┬─────────────────┐
│[37m [0m[01mnumber[0m │[37m [0m[01mstring[0m │[37m [0m[01mboolean[0m │[37m [0m[01mnullValue[0m │[37m [0m[01mundefinedValue[0m │[37m [0m[01m         object[0m │
├────────┼────────┼─────────┼───────────┼────────────────┼─────────────────┤
│[37m [0m[37m    42[0m │[37m [0m[37m  text[0m │[37m [0m[37m   true[0m │[37m [0m[37m         [0m │[37m [0m[37m              [0m │[37m [0m[37m[object Object][0m │
└────────┴────────┴─────────┴───────────┴────────────────┴─────────────────┘"
`;

exports[`General Table Functionality should handle table with no columns but with rows 1`] = `
"┌─────────────┬─────────────┐
│[37m [0m[01mdynamicCol1[0m │[37m [0m[01mdynamicCol2[0m │
├─────────────┼─────────────┤
│[37m [0m[37m     value1[0m │[37m [0m[37m     value2[0m │
└─────────────┴─────────────┘"
`;

exports[`General Table Functionality should maintain data consistency across operations 1`] = `
"┌────────┬────────┐
│[37m [0m[01m  col1[0m │[37m [0m[01m  col2[0m │
├────────┼────────┤
│[37m [0m[37mvalue1[0m │[37m [0m[37mvalue2[0m │
│[37m [0m[37mvalue3[0m │[37m [0m[37mvalue4[0m │
│[37m [0m[37mvalue5[0m │[37m [0m[37mvalue6[0m │
│[37m [0m[37mvalue7[0m │[37m [0m[37mvalue8[0m │
└────────┴────────┘"
`;

exports[`General Table Functionality should support method chaining for all operations 1`] = `
"┌──────┬──────┬──────┐
│[37m [0m[01mcol1[0m │[37m [0m[01mcol2[0m │[37m [0m[01mcol3[0m │
├──────┼──────┼──────┤
│[37m [0m[37m  v1[0m │[37m [0m[37m  v2[0m │[37m [0m[37m  v3[0m │
│[37m [0m[37m  v4[0m │[37m [0m[37m  v5[0m │[37m [0m[37m  v6[0m │
│[37m [0m[37m  v7[0m │[37m [0m[37m  v8[0m │[37m [0m[37m  v9[0m │
└──────┴──────┴──────┘"
`;
